app/models/agents/weather_agent.rb

```diff
@@ -100,14 +107,26 @@ def validate_options
     end
 
     def wunderground
-      Wunderground.new(interpolated['api_key']).forecast_for(location)['forecast']['simpleforecast']['forecastday'] if key_setup?
+      if key_setup?
+        forecast = Wunderground.new(interpolated['api_key'], language: language.upcase).forecast_for(location)
+
+        merged = Hash.new
+        forecast['forecast']['simpleforecast']['forecastday'].each{|daily| merged[daily['period']] = daily}
+
+        txt_forecast = Hash.new
+        merged.each_key do |day|
+            txt_forecast[day] = forecast['forecast']['txt_forecast']['forecastday'].find{|x| x['period'] == day}
+        end
+
+        merged.each_key{|day| merged[day] = merged[day].merge(txt_forecast[day])}
+      end
     end
 
     def forecastio
       if key_setup?
         ForecastIO.api_key = interpolated['api_key']
         lat, lng = location.split(',')
-        ForecastIO.forecast(lat,lng)['daily']['data']
+        ForecastIO.forecast(lat, lng, params: {lang: language.downcase})['daily']['data']
```

Comment:
Yep, I tested it with both data providers and I got back the API response translated to the requested language.